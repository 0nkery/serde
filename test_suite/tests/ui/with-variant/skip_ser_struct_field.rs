#[macro_use]
extern crate serde_derive;

#[derive(Serialize)]
enum Enum {
    #[serde(serialize_with = "serialize_some_other_variant")]
    Struct {
        #[serde(skip_serializing)]
        f1: String,
        f2: u8,
    },
}

fn main() {}
