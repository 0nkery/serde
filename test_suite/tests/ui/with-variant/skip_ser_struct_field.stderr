error: variant `Struct` cannot have both #[serde(serialize_with)] and a field `f1` marked with #[serde(skip_serializing)]
  --> $DIR/skip_ser_struct_field.rs:12:10
   |
12 | #[derive(Serialize)]
   |          ^^^^^^^^^

error: aborting due to previous error

